### DISCUSSION TITLE
Documentation translations

### DISCUSSION THREAD

**Discussion by amercader** [ckan/ckan#7069] [Ideas]
While it's true that CKAN is already used across the world, I think we should aim to have a user base which is as large and diverse as possible. One potential barrier for adoption, specially in the Global South and among small organizations, is having the documentation in English only. We do have a fantastic community of translators that provides the [CKAN interface](https://explore.transifex.com/okfn/ckan/) in multiple languages so perhaps we can set up a system that allows local communities to contribute translations for the documentation or parts of it. OKF can help coordinate a team led by @aivuk that might be able to kick start a Brazilian Portuguese translation and potentially other groups in Latin American for a Spanish translation.

Of course translating documentation comes with an extra set of challenges:

* _Size_: we have [a lot of docs](https://docs.ckan.org/en/2.9/)! It's unlikely that the whole of the documentation can be translated in one go, so we should allow for partial translations (selected pages) and falling back to the English version for untranslated pages (Maybe with a noticed on top saying in the relevant language ("This page is not yet translated. Want to help?")
* _Keeping it up to date_: documentation _is_ code, and we keep ours close to the source code and tied to a particular CKAN version. We update the documentation regularly, and these changes should be ported to any translation.Translations should keep up to date but that of course will depend on each translation team availability. Perhaps translated extensions should include notices stating that these are community-led efforts and might be out of date.
* _Maintenance burden_: after much fiddling we've reached a setup where our docs are transparently updated after merging changes to the main repo. Ideally, translations should not involve extra work from the Tech Team other than initial setup of a language. Translated docs should be maintained by their own communities

There are different approaches followed for supporting different locales for documentation. Projects like [FastAPI](https://github.com/tiangolo/fastapi/tree/master/docs) keep all translations in the same repository, while others like Django use [separate repositories](https://github.com/django/djangoproject.com/tree/main/docs/locale) and [Transifex](https://explore.transifex.com/django/django-docs/).

For a first pilot, I think it would be good to stick to the tools that we use now, namely Read the Docs. Their [multilingual support](https://docs.readthedocs.io/en/stable/localization.html#project-with-multiple-translations) boils down to essentially having one parent project (the current CKAN docs) and multiple separate projects (eg ckan-pt.readthedocs.io) which are marked as translations of the parent one. While projects need to be separated on Transifex that doesn't mean that the sources need to be separated, we could either ask translations maintainers to fork the main ckan repo and build their RTD projects from there, or keep all translations in the main repo (either translated .rst files or potentially with po files) and build separate RTD projects from the same source.

I'm leaning towards keeping it separate at least for now until we test the process and see what it actually involves, but I'm keen on hearing people's thoughts.

Does anyone have experience managing translated documentation for open source projects or similar setups?

**Comment by anicelysantos** [comment]
Hello, 

I'd like to revisit this discussion and express my interest in collaboratively translating the documentation into pt-br. I currently work at the School of Data at Open Knowledge Brasil.
The proposal is to fork the repository and proceed with the translation file by file.
Is it alright to move forward with this proposal?
